ext/fileinfo/fileinfo.stub.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+class finfo
+{
+    function __construct(int $options = FILEINFO_NONE, ?string $arg = null);
+
+    /** @return string|false */
+    function file(?string $file_name = null, int $options = FILEINFO_NONE, $context = null);
+
+    /** @return string|false */
+    function buffer(?string $string = null, int $options = FILEINFO_NONE, $context = null);
+
+    /** @return bool */
```

Comment:
Why is this not a native return type?

Reply:
BC break for anybody extending the `finfo` class

Reply:
Adding return types to the arginfo of methods which did not have a return type formerly, would be a BC break (consider inheritance).

Reply:
@cmb69 Yes I know, just didn't realize how stubs generate stuff. Wonder if there is any value in having this phpDoc here when it's not used anyway?

Reply:
I think it's useful for manual checking, and perhaps somebody likes to use these stub.php files for other purposes as well. And I hope that we can use them sometime as single source of truth – the current situation is awkward.

Reply:
> the current situation is awkward

Agreed. If the stubs were used to generate PHP manual signatures, it would make perfect sense. Right now it's somewhat duplicated and not really tested.